### ISSUE TITLE
Offer addresses from address book contacts as read-only locations in My Places

### ISSUE THREAD

**Issue by PanderMusubi** [osmandapp/Osmand#3026]
Please, offer addresses from address book contacts as read-only in My Places. For people that have well-maintained address book contacts (synced via CardDAV or not) with addresses, it would be very practical to be able to use these via My Places folder called `Contacts`.

In this contacts folder, each contact/address combination should be made available. An address would be a combination of:
* Display name, e.g. "John M. Smith" or "Restaurant Jimmy's"
* Address type, e.g. "Work" or "Private"
* Address, e.g. "123, Sunset Boulevard" or "Dorpstraat 123"
* City, e.g. "New York" or "Amsterdam"
* State/Province, e.g. "NY" or _none_
* Country, e.g. "United States" or "Germany"

The name of the place would be a combination of display name and address type, e.g. "John M. Smith / Work". The address should be parsed to separate the house number from the street name, an approach based on the value of country could ease this. State/province is optional and also largely dependent of the country. If one of the fields is empty (except state/province) the address is not considered to be offered.

Examples from the data above are:
* Category: static, "Contacts"
* Name, required e.g."John M. Smith / Private" or "Restaurant Jimmy's / Work"
* Description: contains of concatenations with new lines of:
  * Country, required
  * State/Province, optional, empty line when n/a
  * Address, required
  * Number, required

Additionally, if enough fields are available but there are some parsing problems, this could be reflected in offering this address non-functional, but only with a warning indicating what should be done in terms of editing the contact in order to improve the usability of that address. Addresses should not be looked up because the availability of certain maps in OsmAnd and of the addresses in the maps can fluctuate.

**TODO:** What needs to be discusses is **when** to go from country, state/province, city, address and number in description to Lat+Lon. Probably each times a new contact/address is found or an update has been detected.

Implementing this would reduce the amount of places where address information is maintained and manually synced. It also reduces the looking up of address while driving, providing a safer use of OsmAnd. This could be developed as an plugin which registers itself in the synchronisation services of Android.

**Comment by Codain** [comment]
Any news on this one? It's not a killer feature but a 'Must have' IMHO. Copying and pasting adresses from the contact book is very painful.

Of course if the authorizations is denier by the user the feature would not be available.

See also #5213

**Comment by stevenroose** [comment]
Yeah I've been wanting/looking for this feature for a while. Google has it out of the box (ofc because it know everything about you). But it's very possible to implement locally and even opt-in by asking to read the user's contacts when they request the feature.

Being able to see my contacts on the map would really be a feature I would use very regularly.

**Comment by gfombell** [comment]
A nice feature (currently missing to Google Maps) would be to have a setting to configure from which zoom level your contacts are shown on the map. 
Besides dreaming of details, I'd like to understand if there is a kind of priority list for future developments and if that "Nice to have" is already ranked.

**Comment by 532910** [comment]
@gfombell, it should be a separate issue:
https://github.com/osmandapp/Osmand/issues/new

**Comment by PanderMusubi** [comment]
Since these are read-only, the icon/symbol used in the favorite can also be derived from the address type:
- home, e.g. the house icon 🏠
- work, e.g. the office buiding icon 🏢
- other, e.g. the triangle flag icon 🚩

For the color, I'd choose a rarely used color such as purple.

By using these icons and color, the read-only favorites stand out from the by default white stars in a yellow circle.

**Comment by stevleibelt** [comment]
I am also supporting this feature request.

Magic Earth has it and it is indeed the killer feature why I am using it too. Just go to the "search" input box and select "from contacts" to hit "use this as destination".
Since my contacts are stored in my nextcloud, google does not know it officially.

Cheers and thanks for your app so far.

**Comment by seniorm0ment** [comment]
Would be really nice to see this.
